code_generation/producer.py

```diff
@@ -25,6 +25,7 @@ def __init__(self, name, call, input, output, scopes):
         self.input = input
         self.output = output
         self.scopes = scopes
+        self.output_checked = False
```

Comment:
I think this attribute is not needed and can be removed. Such tracking is only needed for quantities.

Reply:
yes, i had this initially due to the iterative nature of ProducerGroups, but adding the reserve function for the Group as well does make this obsolete